src/vcpkg/metrics.cpp

```diff
@@ -17,10 +19,92 @@
 #pragma comment(lib, "winhttp")
 #endif
 
+namespace
+{
+    using namespace vcpkg;
+
+    template<typename T>
+    StringLiteral get_metric_name(const T metric, View<MetricEntry<T>> entries)
+    {
+        for (auto&& entry : entries)
+        {
+            if (entry.metric == metric)
+            {
+                return entry.name;
+            }
+        }
+
+        // All metric enums should have corresponding names
+        Debug::println("Error: Metric is missing a name");
+        Checks::exit_fail(VCPKG_LINE_INFO);
+    }
+}
+
 namespace vcpkg
 {
     LockGuarded<Metrics> g_metrics;
 
+    View<MetricEntry<DefineMetric>> Metrics::get_define_metrics()
+    {
+        static constexpr std::array<MetricEntry<DefineMetric>, static_cast<size_t>(DefineMetric::_COUNT)> ENTRIES{{
+            {DefineMetric::AssetSource, "asset-source"},
```

Comment:
An example of why I slightly preferred the string literal form: forgetting an entry in here seems a likely product bug?

Reply:
It is intentional that the build fails if an enum is not defined in this array. However, this only checks for length, so one could put a repeated enum in the array to make it pass. This needs some supporting tests.